libminifi/src/utils/ThreadPool.cpp

```diff
@@ -151,10 +153,10 @@ void ThreadPool<T>::manageWorkers() {
 
   if (nullptr != thread_manager_) {
     while (running_) {
-      auto waitperiod = std::chrono::milliseconds(500);
+      auto wait_period = 500ms;
       {
-        std::unique_lock<std::recursive_mutex> lock(manager_mutex_, std::try_to_lock);
-        if (!lock.owns_lock()) {
+        std::unique_lock<std::recursive_mutex> manager_lock(manager_mutex_, std::try_to_lock);
+        if (!manager_lock.owns_lock()) {
```

Comment:
It looks like this mutex lock should be retried after waiting. Currently it just proceeds to access the shared data without synchronization, after waiting a bit, if the first lock attempt fails.